tests/model_registry/model_catalog/test_model_artifact_search.py

```diff
@@ -0,0 +1,200 @@
+import pytest
+from typing import Self, Any
+
+from ocp_resources.config_map import ConfigMap
+from tests.model_registry.model_catalog.utils import (
+    fetch_all_artifacts_with_dynamic_paging,
+    validate_model_artifacts_match_criteria_and,
+    validate_model_artifacts_match_criteria_or,
+)
+from tests.model_registry.model_catalog.constants import (
+    VALIDATED_CATALOG_ID,
+)
+from kubernetes.dynamic.exceptions import ResourceNotFoundError
+from simple_logger.logger import get_logger
+
+LOGGER = get_logger(name=__name__)
+pytestmark = [pytest.mark.usefixtures("updated_dsc_component_state_scope_session", "model_registry_namespace")]
+MODEL_NAME_ARTIFACT_SEARCH = "RedHatAI/Llama-3.1-8B-Instruct"
+
+
+class TestSearchArtifactsByFilterQuery:
+    @pytest.mark.parametrize(
+        "randomly_picked_model_from_catalog_api_by_source, invalid_filter_query",
+        [
+            pytest.param(
+                {"catalog_id": VALIDATED_CATALOG_ID, "header_type": "registry"},
+                "fake IN ('test', 'fake'))",
+                id="test_invalid_artifact_filter_query_malformed",
+            ),
+            pytest.param(
+                {"catalog_id": VALIDATED_CATALOG_ID, "header_type": "registry"},
+                "filterQuery=ttft_p90.double_value < abc",
+                id="test_invalid_artifact_filter_query_data_type_mismatch",
+            ),
+            pytest.param(
+                {"catalog_id": VALIDATED_CATALOG_ID, "header_type": "registry"},
+                "hardware_type.string_value = 5.0",
+                id="test_invalid_artifact_filter_query_data_type_mismatch_equality",
+            ),
```

Comment:
what do you think about using different models for each of the scenario instead of always `"RedHatAI/Llama-3.1-8B-Instruct"`